test/built-ins/GeneratorPrototype/next/this-val-not-generator.js

```diff
@@ -0,0 +1,69 @@
+// Copyright (C) 2013 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-generator.prototype.next
+es6id: 25.3.1.2
+description: >
+    A TypeError should be thrown from GeneratorValidate (25.3.3.2) if the
+    context of `next` does not defined the [[GeneratorState]] internal slot.
```

Comment:
This pull request is essentially just "miscellaneous improvements," so I'm happy to fix this here.